src/dnsmasq/forward.c

```diff
@@ -1981,14 +2003,28 @@ unsigned char *tcp_request(int confd, time_t now,
 	   /* RFC 6840 5.7 */
 	   if (header->hb4 & HB4_AD)
 	     ad_reqd = 1;
+	  
+	  /* Do this by steam now we're not in the select() loop */
+	  check_log_writer(1); 
+	  
+	  /************ Pi-hole modification ************/
+	  if(piholeblocked)
+	    {
+	      struct all_addr *addrp = NULL;
+	      unsigned int flags = (peer_addr.sa.sa_family == AF_INET) ? F_IPV4 : F_IPV6;
+	      FTL_get_blocking_metadata(&addrp, &flags);
+	      m = setup_reply(header, size, addrp, flags, daemon->local_ttl);
+	      if (have_pseudoheader)
+	        m = add_pseudoheader(header, m, ((unsigned char *) header) + 65536, daemon->edns_pktsz, 0, NULL, 0, do_bit, 0);
+	    }
+	  else
+	  {
+	  /**********************************************/
 	   
 	   /* m > 0 if answered from cache */
 	   m = answer_request(header, ((char *) header) + 65536, (size_t)size, 
 			      dst_addr_4, netmask, now, ad_reqd, do_bit, have_pseudoheader);
 	  
-	  /* Do this by steam now we're not in the select() loop */
-	  check_log_writer(1); 
```

Comment:
Is this deletion intentional?

Reply:
It is not deleted but moved up to line R2008 (see my review comment above).